File: bit_manipulation/power_of_two/Java/PowerOfTwo.java
public class PowerOfTwo {

    public static void main(String[] args) {
	    System.out.println("3 Power of Two? " + isPowerOfTow(3));
        System.out.println("2 Power of Two? " + isPowerOfTow(2));
        System.out.println("365 Power of Two? " + isPowerOfTow(365));
        System.out.println("512 Power of Two? " + isPowerOfTow(512));
        System.out.println("1024 Power of Two? " + isPowerOfTow(1024));
        System.out.println("1738362 Power of Two? " + isPowerOfTow(1738362));
        System.out.println("0 Power of Two? " + isPowerOfTow(0));
    }

    private static boolean isPowerOfTow(int x) {
        return (x & (x-1)) == 0;
    }
}

File: bit_manipulation/power_of_two/JavaScript/power_of_two.js
function isPowerOfTwo(number) {
  return (number && !(number & (number - 1)))
} // Returns a boolean value

var userNumber = 64 // Replace 64 with your own number

if(isPowerOfTwo(userNumber)) console.log(`The number ${userNumber} is a power of TWO`)
else console.log(`The number ${userNumber} is NOT a power of TWO`)

// In your Node environment execute this file as : node power_of_two.js
